paddle/fluid/operators/shape_op.cc

```diff
@@ -58,6 +73,16 @@ Shape Operator.
 
 Return the shape of the input.
 )DOC");
+    AddAttr<bool>("use_mkldnn",
```

Comment:
@lidanqing-intel  Do we still support modifying the properties of native operators now?

Reply:
@baoachun Plan is (as executed in https://github.com/PaddlePaddle/Paddle/pull/36541) that the only new attribute to be added in native ops is "use_mkldnn" . That is becasue in training there are no passes, so we still need an attribute (set with env var FLAGS_use_mkldnn) so that PaddlePaddle runs oneDNN kerenels.

Reply:
`use_mkldnn` attribute is necessary for running mkldnn kernels during training since there are not inference passes in that mode. This attribute is used to determine whether mkldnn kernel should be run or not and that behavior is checked in `GetExpectedKernelType()`. It is a common behavior for all mkldnn kernels and that attribute must be included in original operator version but I can create another onednn-based op and move `mkldnn_data_type` attribute into it, but that cannot be done with `use_mkldnn` attribute. Personally, I would like to keep both of these attributes in native op to allow common inference infrastructure usage but if that's a problem then I'll create a separate op for that.
Here's code for shape op's `GetExpecterKernelType()` which uses `use_mkldnn`  in `this->CanMKLDNNBeUsed()` function https://github.com/PaddlePaddle/Paddle/blob/c4755323b2719692b938e5cb5b6900e2cdd116d3/paddle/fluid/operators/shape_op.cc#L38-L51

Reply:
@baoachun , @jakpiase 
Training does no use passes. So Executor used for training is having a mechanism that when FLAGS_use_mkldnn=True then iteration through all ops is done and those having use_mkldnn attribute are having this attribute to True. So that is the way how during training to let know Paddle to use oneDNN kernels.

Regarding GPU and its "use_cudnn". I looked to code and most of operators (older ones like, conv2d, softmax, gelu, activation, abs,....) are having "use_cudnn" defined. Probably due to API compatibility reasons. 
Some newer ops e.g.  Matmul . Are not having use_cudnn. But matmul CPU and GPU kernel are the same method and execution decision is made based on parameter Place e.g. CPU or GPU.

Some others like affine_channel are having their own kernel for GPU , different from CPU. And decision which run is made based on searching through registers kernels when Key is OpKernelType using following comparison: 

https://github.com/PaddlePaddle/Paddle/blob/1252f4bb3e574df80aa6d18c7ddae1b3a90bd81c/paddle/fluid/framework/op_kernel_type.cc#L55-L60

So you can see that Kernel is chosen if required type matches .e.g library, place, type .  

For GPU kernels there is GPU place. So library plain and GPU place inidicate that GPU kerenels should be used.
For  oneDNN kernels it is a bit harder as there is only CPU place which is used for CPU kernels and oneDNN kernels.
So to pick oneDNN kernels we also need to set library to MKLDNN. And to set library to MKLDNN we currently use "use_mkldnn".  

What I'm trying to write is that if there were oneDNN place then we could do the same as for GPU e.g. discard use_mkldnn attribute. But since there is no oneDNN place and we use CPU place then we need some way to indicate
during the training that library should be MKLDNN and this is done using attribute use_mkldnn.